subgraphs/osmosis/src/common/utils.ts

```diff
@@ -0,0 +1,185 @@
+import { BigInt, ethereum, BigDecimal, cosmos } from "@graphprotocol/graph-ts";
+import {
+  LiquidityPool as LiquidityPoolStore,
+  _TokenPrice,
+} from "../../generated/schema";
+import { MsgPoolParams } from "../modules/Decoder";
+import { PoolFeesType } from "./types";
+import {
+  getOrCreateToken,
+  getOrCreateDexAmmProtocol,
+  getOrCreateLiquidityPoolFee,
+} from "./initializer";
+import * as constants from "../common/constants";
+import * as utils from "../common/utils";
+
+export function bigDecimalToBigInt(n: BigDecimal): BigInt {
+  return BigInt.fromString(n.toString().split(".")[0]);
+}
+
+export function enumToPrefix(snake: string): string {
+  return snake.toLowerCase().replace("_", "-") + "-";
+}
+
+export function prefixID(enumString: string, ID: string): string {
+  return enumToPrefix(enumString) + ID;
+}
+
+export function readValue<T>(
+  callResult: ethereum.CallResult<T>,
+  defaultValue: T
+): T {
+  return callResult.reverted ? defaultValue : callResult.value;
+}
+
+
+export function calculateAverage(items: BigDecimal[]): BigDecimal {
+  let sum = BigDecimal.fromString("0");
+  for (let i = 0; i < items.length; i++) {
+    sum = sum.plus(items[i]);
+  }
+
+  return sum.div(
+    BigDecimal.fromString(BigInt.fromI32(items.length).toString())
+  );
+}
+
+export function updatePoolTVL(
```

Comment:
This would be much cleaner if you encapsulated the token pricing. One thing I have seen others do is generate token prices (lastPriceUSD) within the getOrCreateToken() function. You can even encapsulate the logic further in a getPrice function within the token getter. I would recommend this so that you know you have an update price for the token whenever you get it. Below is an example doing this, and there is some logic to make sure it only updates the price once per block if you grab the token multiple times. 

https://github.com/messari/subgraphs/blob/master/subgraphs/balancer-forks/src/common/initializers.ts#:~:text=%7D-,export%20function%20getOrCreateToken(address%3A%20Address%2C%20blockNumber%3A%20BigInt,%7D,-return%20token%3B

Reply:
Encapsulated the token pricing mechanism into a function at 6903f52ce28319ef61b2eb05dafdabf73a47568c  Thanks!